src/MongoDB/Cursor.c

```diff
@@ -291,6 +291,16 @@ static int php_phongo_cursor_to_array_apply(zend_object_iterator* iter, void* pu
 	return ZEND_HASH_APPLY_KEEP;
 } /* }}} */
 
+static void php_phongo_cursor_id_new_from_id(zval* object, int64_t cursorid TSRMLS_DC) /* {{{ */
```

Comment:
Good catch.

This makes we wonder if we should consider adding serialization support to CursorId, since it is technically just a value object. If so, I think we would just need to always serialize the internal ID as a numeric string for portability across 32-bit and 64-bit architectures. This relates to my other comments about WriteConcern `wtimeout` handling in #1022 (related to #1042). Would you mind opening a new PHPC ticket to record this?

Reply:
Done, see [PHPC-1459](https://jira.mongodb.org/browse/PHPC-1459).